scripts/build-openapi.js

```diff
@@ -1,151 +1,91 @@
+    await sanitizeOAS(outputYamlFile)
+    await circularReferenceCheck(outputYamlFile)
+    await generateReference(outputYamlFile, apiType)
   }
-)
-  .then((gen) => {
-    const oas = new OAS(gen)
-    oas
-      .validate(true)
-      .then(() => {
-        if (!isDryRun) {
-          fs.writeFileSync("./docs/api/admin-spec3.json", gen)
-        }
-      })
-      .catch((err) => {
-        console.log("Error in admin")
-        console.error(err)
-        process.exit(1)
-      })
-  })
-  .catch((err) => {
-    console.log("Error in admin")
-    console.error(err)
-    process.exit(1)
-  })
+}
 
-swaggerInline(
-  [
-    "./packages/medusa/src/models",
-    "./packages/medusa/src/types",
-    "./packages/medusa/src/api/middlewares",
-    "./packages/medusa/src/api/routes/admin",
-  ],
-  {
-    base: "./docs/api/admin-spec3-base.yaml",
-    format: "yaml",
+const generateOASSources = async (outDir, isDryRun) => {
+  const params = ["oas", `--out-dir=${outDir}`]
+  if (isDryRun) {
+    params.push("--dry-run")
   }
-)
-  .then((gen) => {
-    if (!isDryRun) {
-      fs.writeFileSync("./docs/api/admin-spec3.yaml", gen)
-      exec(
-        [
-          "rm -rf docs/api/admin/",
-          "yarn run -- redocly bundle docs/api/admin-spec3.yaml -o docs/api/admin-spec3.yaml --config=docs-util/redocly/config.yaml",
-          "yarn run -- redocly split docs/api/admin-spec3.yaml --outDir=docs/api/admin/",
-        ].join(" && "),
-        (error, stdout, stderr) => {
-          if (error) {
-            throw new Error(`error: ${error.message}`)
-          }
-          console.log(`${stderr || stdout}`)
-          return
-        }
-      )
-    } else {
-      console.log("No errors occurred while generating Admin API Reference")
-    }
+  const { all: logs } = await execa("medusa-oas", params, {
+    cwd: basePath,
+    all: true,
   })
-  .catch((err) => {
-    console.log("Error in admin")
-    console.error(err)
-    process.exit(1)
+  console.log(logs)
+}
+
+const jsonFileToYamlFile = async (inputJsonFile, outputYamlFile) => {
+  const jsonString = await fs.readFile(inputJsonFile, "utf8")
+  const jsonObject = JSON.parse(jsonString)
+  const yamlString = yaml.dump(jsonObject)
+  await fs.writeFile(outputYamlFile, yamlString, "utf8")
+}
+
+const sanitizeOAS = async (srcFile) => {
+  const { all: logs } = await execa(
+    "redocly",
+    [
+      "bundle",
+      srcFile,
+      `--output=${srcFile}`,
+      "--config=docs-util/redocly/config.yaml",
+    ],
+    { cwd: basePath, all: true }
+  )
+  console.log(logs)
+}
+
+const circularReferenceCheck = async (srcFile) => {
+  const parser = new OpenAPIParser()
+  await parser.validate(srcFile, {
+    dereference: {
+      circular: "ignore",
+    },
   })
+  if (parser.$refs.circular) {
+    console.log(`🔴 Unhandled circular references - ${srcFile}`)
+    const circularRefs = [...parser.$refs.circularRefs]
+    circularRefs.sort()
+    console.log(circularRefs)
+  }
```

Comment:
@shahednasser I added a new step to the API docs build script that will output circular references that have not been sanitized/patched. That's how I was able to update `docs-utils/redocly/config.yaml` without going completely insane 🤣